File: examples/workspaces/booksandrecords-spring-boot/src/main/java/com/ms/booksandrecords/processor/factory/CmdProcessorFactory.java
package com.ms.booksandrecords.processor.factory;

import com.ms.booksandrecords.commands.DealCmd;
import com.ms.booksandrecords.processor.CmdProcessor;
import com.ms.booksandrecords.processorImpl.CloseDealProcessor;
import com.ms.booksandrecords.processorImpl.OpenDealProcessor;
import org.springframework.stereotype.Component;

@Component
public class CmdProcessorFactory implements AbstractCmdProcessorFactory <CmdProcessor> {
    @Override
    public CmdProcessor create(DealCmd cmd){
        if(cmd.getClass().getSimpleName().equals("OpenDeal")){
            return new OpenDealProcessor();
        }
        else{
            return new CloseDealProcessor();
        }
    }

}

File: examples/workspaces/booksandrecords-spring-boot/src/main/scala/com/ms/booksandrecords/processor/factory/CmdProcessorFactory.scala
package com.ms.booksandrecords.processor.factory

import com.ms.booksandrecords.commands.{CloseDeal, DealCmd, OpenDeal}
import com.ms.booksandrecords.processor.CmdProcessor
import com.ms.booksandrecords.processorImpl.{CloseDealProcessor, OpenDealProcessor}
import org.springframework.stereotype.Component

@Component class CmdProcessorFactory extends AbstractCmdProcessorFactory[CmdProcessor] {
  override def create(cmd: DealCmd)  =
    cmd match {
      case cmd: OpenDeal => new OpenDealProcessor(cmd)
      case cmd: CloseDeal => new CloseDealProcessor(cmd)
    }
}